src/pages/index.js

```diff
@@ -1,67 +1,128 @@
 import Card from "../components/Card.js";
 import FormValidator from "../components/FormValidator.js";
 import Section from "../components/Section.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import PopupWithForm from "../components/PopupWithForm.js";
+import PopupWithConfirmation from "../components/PopupWithConfirmation.js";
 import UserInfo from "../components/UserInfo.js";
 import {
-  initialCards,
   validationSettings,
-  profileEditForm,
-  profileAddForm,
   profileEditButton,
   profileAddButton,
+  profileAvatarButton,
   profileName,
-  profileDescription,
-  profileInputName,
-  profileInputDescription,
+  profileAbout,
+  profileAvatar,
 } from "../utils/constants.js";
-// import { create } from "lodash";
-// import { profile } from "console";
 
 // FORM VALIDATION INIT
-const profileEditFormValidator = new FormValidator(
-  validationSettings,
-  profileEditForm
-);
-const profileAddFormValidator = new FormValidator(
-  validationSettings,
-  profileAddForm
-);
+const formValidators = {};
+
+const enableValidation = (config) => {
+  const formList = Array.from(document.querySelectorAll(config.formSelector));
+  formList.forEach((formElement) => {
+    const validator = new FormValidator(config, formElement);
+    const formName = formElement.getAttribute("name");
 
-profileEditFormValidator.enableValidation();
-profileAddFormValidator.enableValidation();
+    formValidators[formName] = validator;
+    validator.enableValidation();
+  });
+};
+
+enableValidation(validationSettings);
+
+async function handleSubmit(request, popupInstance, loadingText = "Saving...") {
+  try {
+    popupInstance.renderLoading(true, loadingText);
+    await request();
+    popupInstance.close();
+  } catch (err) {
+    console.error("Error in handleSubmit:", err);
+  } finally {
+    popupInstance.renderLoading(false);
+  }
+}
 
 // USER INFO INIT
 const userInfo = new UserInfo({
   nameSelector: profileName,
-  jobSelector: profileDescription,
+  aboutSelector: profileAbout,
+  avatarSelector: profileAvatar,
 });
 
-function handleEditFormSubmit(data) {
-  userInfo.setUserInfo({
-    name: data.title,
-    job: data.description,
-  });
+let currentUserId = null;
+
+function handleEditFormSubmit(inputValues) {
+  async function makeRequest() {
+    console.log("Updating profile with values:", inputValues);
+    const userData = await api.updateProfile({
+      name: inputValues.name,
+      about: inputValues.about,
+    });
+    userInfo.setUserInfo(userData);
+  }
+  return handleSubmit(makeRequest, profileEditPopup);
+}
 
-  profileEditForm.reset();
-  profileEditFormValidator.disableButton();
-  profileEditPopup.close();
+function handleAddFormSubmit(inputValues) {
+  async function makeRequest() {
+    console.log("Adding card with values:", inputValues);
+
+    const cardData = await api.addCard(inputValues);
+    const cardElement = createCard(cardData, currentUserId);
+    cardSection.addItem(cardElement);
```

Comment:
* New cards should be added only with `prepend` to the start of the section, otherwise the user doesn’t see the new card (if there are **6 or more** cards on the page) and may think that it hasn’t been created


Please, pass `true`

`cardSection.addItem(cardElement, true);`